geomstats/backend/numpy.py

```diff
@@ -269,8 +269,18 @@ def clip(*args, **kwargs):
     return np.clip(*args, **kwargs)
 
 
-def diag(*args, **kwargs):
-    return np.diag(*args, **kwargs)
+def diag(x):
+    x = to_ndarray(x, to_ndim=2)
+    _, n = shape(x)
+    aux = np.vectorize(
+        np.diagflat,
+        signature='(m,n)->(k,k)')(x)
+    k, k = shape(aux)
+    m = int(k / n)
+    result = [
+        aux[i*n:(i+1)*n, i*n:(i+1)*n]
```

Comment:
rule for list comprehension is that if it's too hard to read make it back in a for loop. Here I think it would be better to have this in a for loop.